src/tests/unit/zephyr/common/api-client.test.ts

```diff
@@ -0,0 +1,63 @@
+import { describe, expect, it, vi } from "vitest";
+import { ApiClient } from "../../../../zephyr/common/api-client";
+import { AuthService } from "../../../../zephyr/common/auth-service";
+
+describe("ApiClient", () => {
+  const token = "test-token";
+  const baseUrl = "https://api.zephyrscale.smartbear.com/v2";
+  const apiClient = new ApiClient(token, baseUrl);
+
+  it("should initialize with trimmed baseUrl and default headers", () => {
+    const apiClient = new ApiClient(token, "https://a.url.com/");
+    expect(apiClient.baseUrl).toBe("https://a.url.com");
+    expect(apiClient.defaultHeaders).toEqual(
+      new AuthService(token).getAuthHeaders(),
+    );
+  });
+
+  it("should build correct URL with params", () => {
+    const url = apiClient.getUrl("/projects", { maxResults: 5, startAt: 1 });
+    expect(url).toBe(`${baseUrl}/projects?maxResults=5&startAt=1`);
+  });
+
+  it("should build correct URL with undefined params", () => {
+    const url = apiClient.getUrl("/projects", { maxResults: undefined });
+    expect(url).toBe("https://api.zephyrscale.smartbear.com/v2/projects");
```

Comment:
Here, we should probably use the interpolation with `${baseUrl}`, like in the test case above. Right?

Reply:
That's true! Addressed on bf6c52c4ebceaac4ddcd794b5b6f640962b3227d